packages/worker/src/internals/MainThreadRunner.ts

```diff
@@ -1,23 +1,113 @@
 import fc from 'fast-check';
+import { xorshift128plus } from 'pure-rand';
+import type { IAsyncPropertyWithHooks } from 'fast-check';
 import type { PropertyArbitraries, WorkerProperty } from './SharedTypes.js';
 import { BasicPool } from './worker-pool/BasicPool.js';
 import { Lock } from './lock/Lock.js';
 import type { IWorkerPool, PooledWorker } from './worker-pool/IWorkerPool.js';
 import { OneTimePool } from './worker-pool/OneTimePool.js';
 import { GlobalPool } from './worker-pool/GlobalPool.js';
 
+class CustomAsyncProperty<Ts extends [unknown, ...unknown[]]> implements IAsyncPropertyWithHooks<Ts> {
```

Comment:
To be tested! Could probably directly produce the relevant content for the messages sent to the workers